src/extract.ts

```diff
@@ -496,7 +546,21 @@ export async function extractResult(config: Config): Promise<Benchmark> {
         throw new Error(`No benchmark result was found in ${config.outputFilePath}. Benchmark output was '${output}'`);
     }
 
-    const commit = await getCommit(githubToken);
+    let commit;
+
+    try {
+        commit = getCommitFromWebhookPayload();
+    } catch (err) {
+        if (!githubToken) {
+            if (err instanceof Error) {
+                throw new Error(`${err.message} No github-token provided, could not fallback to GitHub API Request.`);
+            } else {
+                throw err;
+            }
+        }
+
+        commit = await getCommitFromGitHubREST(githubToken);
+    }
```

Comment:
the `getCommit` function contained the majority of this functionality already. Why did you unwrap it here inline rather than modify `getCommit` function

Reply:
Initially I thought it was the most readable, but especially comparing against the upstream `master` branch, I think it's best to put it all behind the `getCommit` function because it accomplishes that purpose. I've moved it there, let me know what you think!